### PR TITLE
[ingress-nginx] Add validation rule for httpsPort parameter

### PR DESCRIPTION
**Pull request by sh1kel**
## Description
Add ingressnginxcontrollers crd validation for httpsPort field. Forbid to use 6443 port for inlet HostPort mode.

## Why do we need it, and what problem does it solve?
Fix issue #3961
inlet HostPort with 6443 port may break k8s controlplane if ingress controller scheduled to control-plane nodes. Validation rule prevents that issue.

## Why do we need it in the patch release (if we do)?

The issue can be reproduced in all previous releases, so it's better to check a port there too.

## What is the expected result?
<pre>
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# ingressnginxcontrollers.deckhouse.io "main" was not valid:
# * spec.hostPort: Invalid value: "object": .spec.httpsPort shouldn't equal to 6443
#
apiVersion: deckhouse.io/v1
kind: IngressNginxController
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"deckhouse.io/v1","kind":"IngressNginxController","metadata":{"annotations":{},"name":"main"},"spec":{"hostPort":{"httpPort":80,"httpsPort":443},"ingressClass":"nginx","inlet":"HostPort","nodeSelector":{"node-role.kubernetes.io/master":""},"tolerations":[{"effect":"NoSchedule","operator":"Exists"}]}}
  creationTimestamp: "2023-11-16T13:04:31Z"
  generation: 1
  name: main
  resourceVersion: "8646"
  uid: 2ccd2801-18ce-4599-8c44-aebd3e279bf7
spec:
  annotationValidationEnabled: false
  chaosMonkey: false
  disableHTTP2: false
  hostPort:
    httpPort: 80
    httpsPort: 6443
    realIPHeader: X-Forwarded-For
  hsts: false
  ingressClass: nginx
  inlet: HostPort
  maxReplicas: 1
  minReplicas: 1
  nodeSelector:
    node-role.kubernetes.io/master: ""
  tolerations:
  - effect: NoSchedule
    operator: Exists
  underscoresInHeaders: false
  validationEnabled: true
</pre>
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: chore
summary: ingressnginxcontrollers crd validation for httpsPort
impact: ingress controller pods will be restarted
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sh1kel. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sh1kel. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sh1kel. I've started the workflow for label 'e2e/run/openstack'...

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 1]
```diff
@@ -203,6 +203,9 @@ spec:
                   anyOf:
                   - {required: ["httpPort"]}
                   - {required: ["httpsPort"]}
+                  x-kubernetes-validations:
+                    - message: .spec.httpsPort shouldn't equal to 6443
```
Comment (konstantin-axenov) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:207]:
```suggestion
                    - message: .spec.httpsPort can't be 6443, it's reserved for kube-apiserver.
```

**Review by konstantin-axenov — APPROVED** [review]

**Pull request merged**